autoload/lsp.vim

```diff
@@ -399,6 +403,22 @@ function! s:ensure_conf(buf, server_name, cb) abort
     call a:cb(l:msg)
 endfunction
 
+let s:file_content = {}
+
+function! s:text_changes(buf) abort
```

Comment:
We can only send diff changes to servers that supports it. Need to look at `syncKind` capabilities. 

https://microsoft.github.io/language-server-protocol/specification#textDocument_didChange

```typescript
/**
 * Defines how the host (editor) should sync document changes to the language server.
 */
export namespace TextDocumentSyncKind {
	/**
	 * Documents should not be synced at all.
	 */
	export const None = 0;

	/**
	 * Documents are synced by always sending the full content
	 * of the document.
	 */
	export const Full = 1;

	/**
	 * Documents are synced by sending the full content on open.
	 * After that only incremental updates to the document are
	 * send.
	 */
	export const Incremental = 2;
}
```

Reply:
You could easily modify this vimscript lang server to support only full and try it out. https://github.com/haya14busa/go-vimlparser/pull/27

Reply:
Thank you, will look it in later.